LCM/scripts/PerformInventory.py

```diff
@@ -111,104 +118,164 @@ def perform_inventory(args):
-
-            # Save the starting timestamp without milliseconds
-            startDateTime = operationStatusUtility.get_current_time_no_ms()
-
-            process = Popen(parameters, stdout = PIPE, stderr = PIPE)
-            stdout, stderr = process.communicate()
-            retval = process.returncode
-
-            printVerboseMessage(stdout)
-
-            # Python 3 returns an empty byte array into stderr on success
-            if stderr == '' or (version_info >= (3, 0) and stderr.decode(encoding = 'UTF-8') == ''):
-                operationStatusUtility.write_success_to_status_file(operation)
-            else:
-                operationStatusUtility.write_failure_to_status_file(operation, startDateTime, stderr)
-                printVerboseMessage(stderr)
-
-            # Combine reports together
-            reportFiles = listdir(dsc_reportdir)
-
-            final_xml_report = '<INSTANCE CLASSNAME="Inventory"><PROPERTY.ARRAY NAME="Instances" TYPE="string" EmbeddedObject="object"><VALUE.ARRAY>'
-            values = []
-            for reportFileName in reportFiles:
-                reportFilePath = join(dsc_reportdir, reportFileName)
-
-                if not isfile(reportFilePath):
-                    continue
-                report = parse(reportFilePath)
-                for valueNode in report.getElementsByTagName('VALUE'):
-                    values.append(valueNode.toxml())
-
-            final_xml_report = final_xml_report + "".join(values) + "</VALUE.ARRAY></PROPERTY.ARRAY></INSTANCE>"
-
-            # Ensure temporary inventory report file permission is set correctly before opening
-            operationStatusUtility.ensure_file_permissions(temp_report_path, '644')
-
-            tempReportFileHandle = open(temp_report_path, 'w')
-            try:
-                tempReportFileHandle.write(final_xml_report)
-            finally:
-                tempReportFileHandle.close()
-
-            # Ensure temporary inventory report file permission is set correctly after opening
-            operationStatusUtility.ensure_file_permissions(temp_report_path, '644')
-
-            system("rm -f " + dsc_reportdir + "/*")
-            move(temp_report_path, report_path)
-
-            # Ensure inventory report file permission is set correctly
-            operationStatusUtility.ensure_file_permissions(report_path, '644')
-        finally:
-            # Release inventory file lock
-            flock(inventorylock_filehandle, LOCK_UN)
+            flock(inventorylock_filehandle, LOCK_EX | LOCK_NB)
+        except IOError:
+            inventorylock_acquired = False
+
+        if inventorylock_acquired:
+            dschostlock_acquired = True
+            # Acquire dsc host file lock
+            if is_oms_config:
+                try:
+                    flock(dschostlock_filehandle, LOCK_EX | LOCK_NB)
+                except IOError:
+                    dschostlock_acquired = False
+
+            if dschostlock_acquired:
```

Comment:
>dschostlock_acquired [](start = 15, length = 20)

We should wait if host is busy with some other operation. otherwise inventory operation may get ignored by frequently running operations. and update management kind of inventory that runs every hour may not get chance for many hours.